balancer/xds/lrs/lrs_test.go

```diff
@@ -172,48 +181,63 @@ func Test_lrsStore_buildStats_rpcCounts(t *testing.T) {
 		name string
 		rpcs []map[internal.Locality]struct {
 			start, success, failure uint64
+			serverData              map[string]float64 // Will be reported with successful RPCs.
 		}
 	}{
 		{
 			name: "one rpcCount report",
 			rpcs: []map[internal.Locality]struct {
 				start, success, failure uint64
+				serverData              map[string]float64
 			}{{
-				localities[0]: {8, 3, 1},
+				localities[0]: {8, 3, 1, nil},
 			}},
 		},
 		{
 			name: "two localities rpcCount reports",
 			rpcs: []map[internal.Locality]struct {
 				start, success, failure uint64
+				serverData              map[string]float64
 			}{{
-				localities[0]: {8, 3, 1},
-				localities[1]: {15, 1, 5},
+				localities[0]: {8, 3, 1, nil},
+				localities[1]: {15, 1, 5, nil},
 			}},
 		},
 		{
 			name: "two rpcCount reports",
 			rpcs: []map[internal.Locality]struct {
 				start, success, failure uint64
+				serverData              map[string]float64
 			}{{
-				localities[0]: {8, 3, 1},
-				localities[1]: {15, 1, 5},
+				localities[0]: {8, 3, 1, nil},
+				localities[1]: {15, 1, 5, nil},
 			}, {
-				localities[0]: {8, 3, 1},
+				localities[0]: {8, 3, 1, nil},
 			}, {
-				localities[1]: {15, 1, 5},
+				localities[1]: {15, 1, 5, nil},
 			}},
 		},
 		{
 			name: "no empty report",
 			rpcs: []map[internal.Locality]struct {
 				start, success, failure uint64
+				serverData              map[string]float64
 			}{{
-				localities[0]: {4, 3, 1},
-				localities[1]: {7, 1, 5},
+				localities[0]: {4, 3, 1, nil},
+				localities[1]: {7, 1, 5, nil},
 			}, {
-				localities[0]: {0, 0, 0}, // This is shouldn't cause an empty report for locality[0].
-				localities[1]: {1, 1, 0},
+				localities[0]: {0, 0, 0, nil}, // This is shouldn't cause an empty report for locality[0].
+				localities[1]: {1, 1, 0, nil},
+			}},
+		},
+		{
+			name: "two localities rpcCount reports with server loads",
+			rpcs: []map[internal.Locality]struct {
+				start, success, failure uint64
+				serverData              map[string]float64
+			}{{
+				localities[0]: {8, 3, 1, map[string]float64{"cpu": 15, "mem": 20}},
+				localities[1]: {15, 4, 5, map[string]float64{"net": 5, "disk": 0.8}},
 			}},
```

Comment:
Add a test case that contains more than one map item, so we can see what happens to multiple serverData case?

Reply:
Each RPC reports only one map (it's the trailer).

And in this test, the same map is reported for each successful RPC, so what you want is already covered?

Reply:
This is the only test that contains server data, and both localities get update once. What I mean is to have another test, so that multiple rpcs update the serverData for the same locality, so we can verify the update logic for serverData is correct.

Reply:
Done